src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +378,313 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+                    v2Region.StartColumn > 0 ||
+                    v2Region.EndColumn > 0 ||
+                    v2Region.CharOffset > 0 ||
+                    v2Region.CharLength > 0)
+                {
+                    if (v2Region.StartLine > 0)
+                    {
+                        // The start of the region is described by line/column
+                        region.StartLine = v2Region.StartLine;
+                        region.StartColumn = v2Region.StartColumn > 0
+                            ? v2Region.StartColumn
+                            : 1;
+                    }
+                    else
+                    {
+                        // The start of the region is described by character offset
+
+                        if (v2Region.CharOffset == 0)
+                        {
+                            region.Offset = 0;
+                        }
+                        else
+                        {
+                            // Try to get the byte offset using the file encoding and contents
+                            region.Offset = GetRegionByteOffset(v2Region, uri);
+                        }
+                    }
+
+                    if (v2Region.CharLength > 0)
+                    {
+                        // The end of the region is described by character length
+                        // Try to get the byte length using the file encoding and contents
+                        region.Length = GetRegionByteLength(v2Region, uri);
+                    }
+                    else if (v2Region.EndLine > 0)
+                    {
+                        region.EndLine = v2Region.EndLine;
+
+                        if (v2Region.EndColumn > 0)
+                        {
+                            region.EndColumn = v2Region.EndColumn;
+                        }
+                        else
+                        {
+                            // Try to get the end column using the file encoding and contents
+                            region.EndColumn = GetRegionEndColumn(v2Region, uri);
+                        }
+                    }
+                    else if (v2Region.EndColumn > 0)
+                    {
+                        region.EndColumn = v2Region.EndColumn;
+                    }
+                    else
+                    {
+                        // THIS IS A PROBLEM. IF ALL THE "END" PROPERTIES ARE MISSING,
+                        // IT MEANS "THE REST OF THE StartLine". BUT IF CHARLENGTH IS
+                        // PRESENT BUT 0, IT MEANS "INSERTION POINT". AND WE CAN'T TELL
+                        // THE DIFFERENCE.
+                        // TODO: Issue #932
+
+                        // Assume it's an insertion point
+                        if (v2Region.StartLine > 0)
+                        {
+                            region.EndLine = region.StartLine;
+                            region.EndColumn = region.StartColumn;
```

Comment:
This works only if the start of the v2 region was specified by `startLine`/`startColumn` If the start of the v2 region was specified by `charOffset`, then you never populated `region.startLine` and `region.startColumn`.

But good news! The v1 spec says:
> If none of endLine, endColumn, or length is present, the length of the region shall be considered to be 0.

So, if you want an insertion point, just set `region.EndLine = region.EndColumn = region.Length = 0`. You don't need the conditonal on `v2Region.StartLine` being present. And then you don't need the helper method `GetRegionStart`.

Reply:
Ah gotcha

Reply:
Fixed